### DISCUSSION TITLE
Quey in running Best Model Selection

### DISCUSSION THREAD

**Discussion by saleharashid01** [AammarTufail/python_ka_chilla2023#81] [Q&A]
![saq](https://user-images.githubusercontent.com/121869148/213816546-a03acb62-39db-470a-95af-49b4afe3a215.jpg)
when I assign variables X and y we use double square brackets for both X and y, it is working till training the model. 
But i enter the code for the best model and run it, the error says that y should be 1 d. as shown in image. i set single bracket to y but still its not running.  what should i do
![sa](https://user-images.githubusercontent.com/121869148/213816437-3c0c683c-5f9a-4fa5-8f53-abe6922bebda.jpg)
![he](https://user-images.githubusercontent.com/121869148/213816450-37cfcb57-6581-4aa8-9e86-b3889aae62af.jpg)

**Comment by bakhtawar-ishaq** [comment]
Y k around single bracket lagany k bad code ko dobara run krein takkay data update ho jaye.

**Reply by saleharashid01** [reply]
it didn't work either

**Reply by Muhammad-Ali-Butt** [reply]
apko y aur X ki shape theek karni ho gi  : X.shape kr k aur y.shape kr k output check krna k same aya for example : agar X ki shape (847, 6) aur y ki shape (847, ) ho gi tu chalay ga matlab ye k y ki shape mai comma k baad koi number show na ho like this (847, ) aur rows same ho X k, in the case of my example dono ki rows same ha 847 aur X mai columns 6 ha. 

agar error na resolve ho tu pooray code ka ss bhejna shuru se

**Reply by saleharashid01** [reply]
got the issue, when I was applying train_test-split, 
i was mentioning X_train, y_tarain, X_test, y_test while i should use X_train, X_test, y_train,y_test =


it worked for me,Jazak Allah
